src/uwtools/tests/drivers/test_jedi.py

```diff
@@ -0,0 +1,207 @@
+"""
+JEDI driver tests.
+"""
+import datetime as dt
+import logging
+from pathlib import Path
+from unittest.mock import DEFAULT as D
+from unittest.mock import Mock, call, patch
+
+import yaml
+from iotaa import asset, external
+from pytest import fixture
+
+from uwtools.config.formats.yaml import YAMLConfig
+from uwtools.drivers import jedi
+from uwtools.scheduler import Slurm
+from uwtools.tests.support import regex_logged
+
+# Fixtures
+
+
+@fixture
+def cycle():
+    return dt.datetime(2024, 2, 1, 18)
+
+
+# Driver fixtures
+
+
+@fixture
+def config(tmp_path):
+    return {
+        "jedi": {
+            "execution": {
+                "batchargs": {
+                    "export": "NONE",
+                    "nodes": 1,
+                    "stdout": "/path/to/file",
+                    "walltime": "00:02:00",
+                },
+                "envcmds": [
+                    "module load some-module",
+                    "module load jedi-module",
+                ],
+                "executable": "/path/to/qg_forecast.x",
+                "mpiargs": ["--export=ALL", "--ntasks $SLURM_CPUS_ON_NODE"],
+                "mpicmd": "srun",
+            },
+            "configuration_file": {
+                "base_file": str(tmp_path / "jedi.yaml"),
+                "update_values": {"baz": "qux"},
+            },
+            "files_to_copy": {
+                "foo": "/path/to/foo",
+                "bar/baz": "/path/to/baz",
+            },
+            "files_to_link": {
+                "foo": "/path/to/foo",
+                "bar/baz": "/path/to/baz",
+            },
+            "run_dir": str(tmp_path),
+        },
+        "platform": {
+            "account": "me",
+            "scheduler": "slurm",
+        },
+    }
+
+
+@fixture
+def config_file(config, tmp_path):
+    path = tmp_path / "config.yaml"
+    with open(path, "w", encoding="utf-8") as f:
+        yaml.dump(config, f)
+    return path
+
+
+@fixture
+def driverobj(config_file, cycle):
+    return jedi.JEDI(config=config_file, cycle=cycle, batch=True)
+
+
+# Driver tests
+
+
+def test_JEDI(driverobj):
+    assert isinstance(driverobj, jedi.JEDI)
+
+
+def test_JEDI_dry_run(config_file, cycle):
+    with patch.object(jedi, "dryrun") as dryrun:
+        driverobj = jedi.JEDI(config=config_file, cycle=cycle, batch=True, dry_run=True)
+    assert driverobj._dry_run is True
+    dryrun.assert_called_once_with()
+
+
+def test_JEDI_files_copied(driverobj):
+    with patch.object(jedi, "filecopy") as filecopy:
+        driverobj._driver_config["run_dir"] = "/path/to/run"
+        driverobj.files_copied()
+        assert filecopy.call_count == 2
+        assert (
+            call(src=Path("/path/to/baz"), dst=Path("/path/to/run/bar/baz"))
+            in filecopy.call_args_list
+        )
+        assert (
+            call(src=Path("/path/to/foo"), dst=Path("/path/to/run/foo")) in filecopy.call_args_list
+        )
```

Comment:
Can you explain why the `files_copied()` and `files_linked()` tests are different for JEDI than the other drivers?

Reply:
Since @NaureenBharwaniNOAA and I paired on this, and I recommended these simpler tests, I can try to answer.

Honestly, I'm asking myself why the other tests are so complex! I think I probably wrote the one for FV3, from which MPAS and MPAS Init probably borrowed, and I'm wondering why I didn't use mocks there like we've done here, since the `filecopy` and `symlink` task methods from `uwtools.utils.tasks`, which perform the actual copy/link actions, have their own tests.

I wonder now if I wrote the FV3 tests before `filecopy` and `symlink` appeared in `uwtools.utils.tasks`, such that the FV3 tests _had to_ test the real filesystem operations, because the FV3 driver was initially performing them directly rather than delegating to `filecopy` and `symlink`.

I'm looking at the FV3 tests and haven't convinced myself yet that these couldn't be simplified with mocks. What do you think?

Reply:
That would make sense to me. To be honest, I had previously chalked these up to be "Paul loves to `parametrize` his unittests," but perhaps I could have put some extra thought into that before copying them over to other drivers. I have a strong preference for these tests. We should carry this change over to the other drivers at some point.

Reply:
I might still parametrize these two, given how similar they are, though I'm not asking for that as a reviewer. Even parametrized, they'd still be much simpler than the existing FV3/MPAS/MPASInit versions.